src/components/search/FullSiteSearch.jsx

```diff
@@ -0,0 +1,131 @@
+/** @jsxImportSource @emotion/react */
+import tw from "twin.macro";
+import React from "react";
+import { XIcon } from "../icons/Icons";
+import axios from "axios";
+
+const SearchResultItem = ({ placeTitle, placeLink, dealTitle, className }) => {
+  return (
+    <div className={className} tw="p-4">
+      <a href={`/place/${placeLink}`}>
+        <h3 tw="font-bold text-xl">{placeTitle}</h3>
+        <p>{dealTitle}</p>
+      </a>
+    </div>
+  );
+};
+export const FullSiteSearch = ({ showSearch, onSearchDismiss }) => {
+  const [searchResults, setSearchResults] = React.useState(null);
+  const [searchTerm, setSearchTerm] = React.useState("");
+
+  const fetchData = React.useCallback(async () => {
+    if (searchTerm.length < 2) {
+      setSearchResults(null);
+      return;
+    }
+    try {
+      const res = await axios.get(
```

Comment:
We should use a debouncer here. Delay the requests otherwise we're going to be sending multiple requests at once.